File: qa/esb/java/test_framework/test_cases/for_windows/mediation/transformation/scripting/java_script/referenced_script/resources/addNumTransform.js
<x><![CDATA[
  function transformRequest(mc) {
     var x = mc.getPayloadXML()..*::x.toString();
     var y = mc.getPayloadXML()..*::y.toString();
     mc.setPayloadXML(
        <ns1:add xmlns:ns1="http://add.math.com">
			<ns1:x>{x}</ns1:x>
			<ns1:y>{y}</ns1:y>
        </ns1:add>);
  }

  function transformResponse(mc) {
     var response = mc.getPayloadXML()..*::response.toString();
     mc.setPayloadXML(
        <ns1:addResponse xmlns:ns1="http://add.math.com">
			<ns1:return>{response}</ns1:return>
        </ns1:addResponse>);
  }
]]></x>

File: qa/esb/java/test_framework/test_cases/for_windows/mediation/transformation/scripting/ruby/referenced_script/resources/addNumTransform.rb
<x><![CDATA[
require 'rexml/document'
include REXML

def transformRequest(mc)
  newRequest= Document.new '<m:add xmlns:m="http://add.math.com">'<<
            '<m:x></m:x><m:y></m:y></m:add>'

  newRequest.root.elements[1].text = mc.getPayloadXML().root.elements[1].get_text
  newRequest.root.elements[2].text = mc.getPayloadXML().root.elements[1].get_text

  mc.setPayloadXML(newRequest)

end

def transformResponse(mc)
  newResponse = Document.new '<m:addResponse xmlns:m="http://add.math.com"><m:return>' <<
    '</m:return></m:addResponse>'

  puts(mc)
  puts(newResponse)

  response=mc.getPayloadXML().root.get_elements('ns:return')
 
  puts(mc.getPayloadXML())
  puts(mc.getPayloadXML().root)
  puts(mc.getPayloadXML().root.get_elements('ns:return'))
  puts(response)

  newResponse.root.elements[1].text = response[0].get_text

  mc.setPayloadXML(newResponse)

end
]]></x>
